vassal-app/src/main/java/VASSAL/build/module/GameState.java

```diff
@@ -746,15 +746,11 @@ else if (END_SAVE.equals(theCommand)) {
   public void saveGame(File f) throws IOException {
 // FIXME: Extremely inefficient! Write directly to ZipArchive OutputStream
     final String save = saveString();
-    try (FastByteArrayOutputStream ba = new FastByteArrayOutputStream()) {
-      try (OutputStream out = new ObfuscatingOutputStream(ba)) {
+    try (FileArchive archive = new ZipArchive(f)) {
+      try (OutputStream out = new ObfuscatingOutputStream(archive.getOutputStream(SAVEFILE_ZIP_ENTRY))) {
```

Comment:
Won't we fail to close the `OutputStream` from the archive if the `OOS` ctor throws?

Shouldn't it be this instead?

```
try (OutputStream zout = archive.getOutputStream(SAVEFILE_ZIP_ENTRY);
     OutputStream out = new ObfuscatingOutputStream(zout)) {
```

Reply:
Both should work but the one with explicit declaration of the stream variables looks safer.